### ISSUE TITLE
Changes window order on restart

### ISSUE THREAD

**Issue by JeffreySmalley** [mikl/browser-go-to-random-tab#6]
since the latest update, when i have this enabled, all the window orders change randomly on session restore.

**Comment by mikl** [comment]
Hi @JeffreySmalley, I’m not sure what you mean by window order. Could you describe the problem a little more in detail?

**Comment by JeffreySmalley** [comment]
When closing firefox(should have said that earlier) with this extension active, if using "Open previous windows and tabs" in the Firefox general settings, the windows (except the first window) will are open in a random order.

Which changes their order in the taskbar

**Comment by mikl** [comment]
Ok, I guess this is on Windows then?

I’m afraid I don’t see how this extension could cause Firefox’ session restoration to work differently.

The extension’s code only runs when triggered by the toolbar button or the new keyboard shortcut, so it should be impossible for it to affect session restoration.

You could try downloading a previous version of the extension from here: https://addons.mozilla.org/en-GB/firefox/addon/go-to-random-tab/versions/

If nothing else, it would be useful to know if it was the 2.1.0 update or the 2.1.1 update that somehow triggers this change.

**Comment by JeffreySmalley** [comment]
Yes windows, also can confirm that 2.1.0 has no issue, while 2.1.1 does have the issue, even after purging the extension and reinstalling it.


Edit, went through all the code, and have no clue what could possibly be the cause, since nothing really even changes between the versions that could do this.

Will try again in the next build, maybe there is a bug, not with the code but something random happened and the next build will fix it. Dunno what to say, just that 2.1.1 consistently causes an issue every launch, and 2.1.0 does not cause it at all.

**Comment by mikl** [comment]
Yeah, you could argue that this is a bug in Firefox itself. Since [the only changes in v2.1.1](https://github.com/mikl/browser-go-to-random-tab/compare/d5529b2...v2.1.1) are:

1. Reworking the icon set-up with support for dark themes
2. Setting the default area for the toolbar icon to be the tab strip (makes sense, since it’s a button for navigating tabs).
3. Fixing a deprecation warning in the manifest (`applications` is now `browser_specific_settings`).

None of this should affect the window order. If it does, there’s probably a deeper issue somewhere.
